### ISSUE TITLE
Android App Does NOT Upload Video

### ISSUE THREAD

**Issue by chairstacker** [nextcloud/android#3190]
### Actual behaviour
The app only uploads photos, i.e. jpg files from my 'Auto upload' folders.
### Expected behaviour
The app should upload at least video, i.e. mp4, files as well.
 
### Steps to reproduce
1. Set Auto Upload Folder
2. Take a video
3. Check Nextcloud server => it isn't there

### Environment data
Android version:
Android 8.0.0

Device model: 
Samsung Galaxy S7

Stock or customized system:
Stock software for AT&T version (G930A)

Nextcloud app version:
Latest, i.e #.3.0

Nextcloud server version:
14.0.1

#### Web server error log
```not a server issue

Insert your webserver log here
```not a server issue


#### Nextcloud log (data/nextcloud.log)
```not a server issue

Insert your Nextcloud log here
```not a server issue

**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by chairstacker** [comment]
Nope, not related to any of them - these issues are either way too old to be still relevant or describe something completely different.

**Comment by tobiasKaminsky** [comment]
@chairstacker AutoUpload distinguishes between video and image, have you set it up properly for videos? You see a video icon in front of Camera name.

![2018-10-26-075242](https://user-images.githubusercontent.com/5836855/47547059-35246e80-d8f4-11e8-846c-668b82b1da08.png)

**Comment by chairstacker** [comment]
Good point - this seems to be a nicely hidden change from the way it was done before.
Could find this information anywhere. 
As ever, a little bit of a documentation would go a long way.
Thanks for your help, highly appreciated.

**Comment by chairstacker** [comment]
Maybe, if it said something like images or pictures / photos only from this folder where the image icon is shown in the app it would have made me look further down for the camera icon.
And creating a forum entry that has the words video and upload in the subject line might help as well. A quick mention of the need for separate entries in the upload list for images and videos might address that, too.

**Comment by tobiasKaminsky** [comment]
Going away from our app to explain something is a no go in my opinion.

@nextcloud/designers @AndyScherzinger do you have a good idea?

Maybe we need a little hint, like when no files are in a folder: new folders can be configured once the app detects media images/videos in it.

**Comment by jancborchardt** [comment]
Hm, what exactly is the issue in the flow here? It does upload images, but not videos? Aren’t they done by the same app (Camera) and should hence both work?

(This should not need any configuration or explanation but simply work.)

**Comment by denis-pujdak-adm-it** [comment]
@jancborchardt I agree with you.

By the way because my videos not being uploaded I removed predefined Auto Upload for photos and videos and added a custom folder configuration and now all files are uploaded from the Camera folder with no problems.

![Screenshot_20190827-222710_Nextcloud](https://user-images.githubusercontent.com/13495631/63809484-4fffaf00-c91a-11e9-9979-e1de04f71932.png)

**Comment by cowbe0x004** [comment]
This is still not working for me, tried both predefined video folder (the same one as pictures in /Camera) and manually created custom folder. I'm on Android 10, app version 3.8.1.

**Comment by neufeind** [comment]
Worked for me once I had a video there and could activat the uploading.

But how about not configuring "what to upload" separately but as part of the properties of a folder? You'd be able to edit a folder and right there where you can select "only on wireless" you could also whitelist what to actually upload.
One edgecase that I could think of would be that people might want to upload images immediately but videos only via wifi? Hmm, but in that case, could they add a manual folder-entry for the same folder and on one set it up to only work on images and the other to only care about videos (where you'd have videos synced only via wifi).

**Comment by riggs-** [comment]
How is this still an open issue after almost 4 years? The current behaviour makes no sense at all. If someone selects a folder for auto-upload, the whole folder should be synchronised, the file type should not matter.

**Comment by tobiasKaminsky** [comment]
The idae back then was that an user might want to have direct auto upload of images, as they tend to be small, and auto upload big videos only via wifi.
Maybe this is nowayads with big data plans obsolete…

What do you think? @jancborchardt @AlvaroBrey

**Comment by AlvaroBrey** [comment]
> The idae back then was that an user might want to have direct auto upload of images, as they tend to be small, and auto upload big videos only via wifi. Maybe this is nowayads with big data plans obsolete…
> 
> What do you think? @jancborchardt @AlvaroBrey

I think uploading one but not the other makes sense, but I wouldn't have them separated in two different sections for the same folder. For me this should be an option within the sync settings for that folder.

**Comment by jancborchardt** [comment]
Indeed makes sense to also include videos in auto-upload. There's already the "only upload on Wifi" option and that should be enough for now if data is a concern, so let's keep it simple and not add to that.
